runtime-sdk/modules/contracts/src/abi/oasis/storage.rs

```diff
@@ -166,29 +166,42 @@ fn get_instance_store<'a, C: Context>(
     // Determine which store we should be using.
     let store_kind: StoreKind = store_kind.try_into().map_err(|_| wasm3::Trap::Abort)?;
 
-    Ok(store::for_instance(
-        ec.tx_context,
-        ec.instance_info,
-        store_kind,
-    )?)
+    let instance_store = store::for_instance(ec.tx_context, ec.instance_info, store_kind);
```

Comment:
Should the crypto functions also be updated to return nicer errors (e.g. if invalid input parameters are provided). Can also be done in a later PR.

Reply:
Good idea, will add those as well.